### ISSUE TITLE
gtk+ failing to install

### ISSUE THREAD

**Issue by soredake** [edannenberg/kubler#33]
http://pastebin.com/BJsrRuLF

https://github.com/gentoo/gentoo/blob/d321ab3053915b22883a6681cead236190b994e6/eclass/gnome2-utils.eclass#L450 pointing to $EPREFIX inster of $ROOT, and so it fails.

i tried this in configure_rootfs and configure_bob:
`sed -i -e 's|updater=${EPREFIX}/usr/bin/gtk-query-immodules-2.0|/emerge-root/usr/bin/gtk-query-immodules-2.0|g' /usr/portage/eclass/gnome2-utils.eclass`
but it still pointing to /usr/bin/gtk-query-immodules-2.0 and fails, maybe this code not changing eclass.
Perhaps you can tell how to solve this problem?

**Comment by edannenberg** [comment]
Looks like it should work to me. Gonna take a closer look later today.

**Comment by soredake** [comment]
Here is my https://transfer.sh/2JB1Y/buildconfig.sh buildconfig that fails to install gtk+

**Comment by edannenberg** [comment]
Thanks, that made things a little easier.

I tried to execute `gtk-query-immodules-2.0` manually in an interactive build container after the failed install. More errors down the road as dependecies of the binary are not in the system path. 

You will usually run into a wall with packages that execute some package specific binaries as part of the build process. Had the same issue when trying to install `tomcat-native` recently. I resolved to some trickery as I found all you really need is a ready binary package and the install will run fine for custom `ROOT` installs. Try the following:

    PACKAGES="=x11-libs/gtk+-2.24.29"

    configure_bob()
    {
    update_use x11-libs/gdk-pixbuf +X
    update_use x11-libs/cairo +X
    emerge =x11-libs/gtk+-2.24.29
    }

So just install the troublesome package in the build container in the `configure_bob` hook, not what we want but it creates a binary package which is then picked up in the next phase when `PACKAGES` are installed at the custom root.

I should probably document that. :p
